pkg/codec/map_decoder.go

```diff
@@ -0,0 +1,246 @@
+package codec
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"reflect"
+	"strconv"
+
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/types"
+)
+
+func DecoderFromMapDecoder(decoder types.MapDecoder, extraHooks ...mapstructure.DecodeHookFunc) (types.Decoder, error) {
+	if decoder == nil {
+		return nil, errors.New("decoder must not be nil")
+	}
+
+	numExtraHooks := len(extraHooks)
+	hooks := make([]mapstructure.DecodeHookFunc, numExtraHooks+2)
+	copy(hooks, extraHooks)
+	hooks[numExtraHooks] = SliceToArrayVerifySizeHook
+	hooks[numExtraHooks+1] = BigIntHook
+	return &mapDecoder{decoder: decoder, hooks: hooks}, nil
```

Comment:
Should we return the concrete type here instead of the interface?
```suggestion
func DecoderFromMapDecoder(decoder types.MapDecoder, extraHooks ...mapstructure.DecodeHookFunc) (*MapDecoder, error) {
	if decoder == nil {
		return nil, errors.New("decoder must not be nil")
	}

	numExtraHooks := len(extraHooks)
	hooks := make([]mapstructure.DecodeHookFunc, numExtraHooks+2)
	copy(hooks, extraHooks)
	hooks[numExtraHooks] = SliceToArrayVerifySizeHook
	hooks[numExtraHooks+1] = BigIntHook
	return &MapDecoder{decoder: decoder, hooks: hooks}, nil
```

Reply:
map decoder has been completely removed